tests/components/home_connect/test_sensor.py

```diff
@@ -1,89 +1,158 @@
-    await hass.async_block_till_done()
-
-    assert hass.states.is_state(
-        "sensor.dishwasher_remaining_program_time", "unavailable"
-    )
-
-    hc_app.status.update({"BSH.Common.Option.RemainingProgramTime": {"value": 0}})
-    dispatcher_send(hass, SIGNAL_UPDATE_ENTITIES, hc_app.haId)
-    await hass.async_block_till_done()
 
-    assert hass.states.is_state(
-        "sensor.dishwasher_remaining_program_time", "unavailable"
-    )
 
-    hc_app.status.update(
-        {
-            "BSH.Common.Status.OperationState": {
-                "value": "BSH.Common.EnumType.OperationState.Run"
-            }
-        }
-    )
-
-    dispatcher_send(hass, SIGNAL_UPDATE_ENTITIES, hc_app.haId)
-    await hass.async_block_till_done()
-
-    assert not hass.states.is_state(
-        "sensor.dishwasher_remaining_program_time", "unavailable"
+@pytest.mark.parametrize("appliance", [TEST_HC_APP], indirect=True)
+async def test_event_sensors(
+    appliance,
+    bypass_throttle,
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    integration_setup: Callable[[], Awaitable[bool]],
+    setup_credentials: None,
+    get_appliances: MagicMock,
+) -> None:
+    """Tests for sensors that are only available after an event happens."""
+    EVENT_RUN = (
+        EVENT_PROG_DELAYED_START,
+        EVENT_PROG_REMAIN_NO_VALUE,
+        EVENT_PROG_RUN,
+        EVENT_PROG_UPDATE_1,
+        EVENT_PROG_UPDATE_2,
+        EVENT_PROG_END,
     )
 
-    hc_app.status.update({"BSH.Common.Option.RemainingProgramTime": {"value": 3600}})
+    ENTITY_ID_STATES = [
+        (
+            "sensor.dishwasher_operation_state",
+            ["Delayed", "Delayed", "Run", "Run", "Run", "Ready"],
+        ),
+        (
+            "sensor.dishwasher_remaining_program_time",
+            [],
+        ),
+        (
+            "sensor.dishwasher_program_progress",
+            [
+                "unavailable",
+                "unavailable",
+                "60",
+                "80",
+                "99",
+                "99",
+            ],
+        ),
+    ]
+    appliance.status.update(EVENT_PROG_END)
+
+    get_appliances.return_value = [appliance]
 
-    dispatcher_send(hass, SIGNAL_UPDATE_ENTITIES, hc_app.haId)
-    await hass.async_block_till_done()
+    assert config_entry.state == ConfigEntryState.NOT_LOADED
+    assert await integration_setup()
+    assert config_entry.state == ConfigEntryState.LOADED
 
     assert hass.states.is_state(
         "sensor.dishwasher_remaining_program_time", "unavailable"
     )
+
+    for count, event in enumerate(EVENT_RUN):
+        appliance.status.update(event)
+        for entity_id, states in ENTITY_ID_STATES:
+            await async_update_entity(hass, entity_id)
+            await hass.async_block_till_done()
+            # Check to see if sensor is unavailable or a datetime object.
+            if entity_id == "sensor.dishwasher_remaining_program_time":
```

Comment:
We want to avoid conditions in tests. Please either try to parametrize the tests by using `pytest.mark.parametrize` or write separate tests to remove the conditions.

Eg here I'd try to make parameters for `event_run`, `entity_id` and `state`.